src/Microsoft.AspNetCore.Blazor/Routing/Router.cs

```diff
@@ -74,15 +79,19 @@ protected virtual void Render(RenderTreeBuilder builder, Type handler, IDictiona
             builder.CloseComponent();
         }
 
-        private void Refresh()
+        private void Refresh(bool useFallback = false)
         {
-            var locationPath = UriHelper.ToBaseRelativePath(_baseUri, _locationAbsolute);
+            var locationPath = useFallback ? FallbackRoute : UriHelper.ToBaseRelativePath(_baseUri, _locationAbsolute);
             locationPath = StringUntilAny(locationPath, _queryOrHashStartChar);
             var context = new RouteContext(locationPath);
             Routes.Route(context);
             if (context.Handler == null)
             {
-                throw new InvalidOperationException($"'{nameof(Router)}' cannot find any component with a route for '/{locationPath}'.");
+                if (useFallback || FallbackRoute == null)
```

Comment:
I know this is a really minor point, but for code style consistency, could we have braces around line 91?

Reply:
Changed and committed.